### PR TITLE
Switch over to use containerized clusters on CI

### PR DESCRIPTION
**Pull request by rmohr**
* Use k8s-1.9.3 instead of vagrant
* Make sure that all components select random ports during the whole testing phase, to allow parallel builds
* Make sure that each build slot only keeps tags for the latest 5 builds. Older builds are untagged and deleted.
* If /dev/kvm is present in the container, use it. Otherwise fall back to emulation.

One TODO  remaining: Detect if the docker registry volume reaches a critical size and delete it, to avoid fillling up the whole space.

### CODE REVIEW

**Comment by stu-gott** [comment]
I like the direction this is taking us!

**Comment by cynepco3hahue** [comment]
lgtm

**Comment by rmohr** [comment]
retest this please

**Comment by rmohr** [comment]
@fabiand @cynepco3hahue this PR should now be ready (lets wait for the ci results). @cynepco3hahue could you do another review? @fabiand There is a temporary /dev/kvm fix included which will use /dev/kvm if present and fall back to emulation otherwise. I want to add a proper fix in #804. Is that fine for you?

File: cluster/ephemeral-provider-common.sh [hunk 1]
```diff
@@ -2,24 +2,48 @@
...
 docker_tag=devel
 kubeconfig=${BASE_PATH}/cluster/$PROVIDER/.kubeconfig
-docker_prefix=localhost:5000/kubevirt
+docker_prefix=localhost:$(_port registry)/kubevirt
 manifest_docker_prefix=registry:5000/kubevirt
 EOF
 }
 
+function _registry_volume() {
+    if [ -n "${JOB_NAME}" ]; then
```
Comment (davidvossel) [inline cluster/ephemeral-provider-common.sh]: There's a lot of branching logic around "JOB_NAME". If JOB_NAME then do this, if not do this. 

I'd rather see a single code path and have JOB_NAME something that is always present regardless of CI or local dev.

File: cluster/ephemeral-provider-common.sh [hunk 2]
```diff
@@ -2,24 +2,48 @@
...
+_prefix=${_prefix}${EXECUTOR_NUMBER}
+
+_cli='docker run --privileged --rm -v /var/run/docker.sock:/var/run/docker.sock kubevirtci/cli@sha256:6d7380015a2743992a1b3e61dfe4e79925e825e40a93e4594c125822d36c56fc'
 
 function _main_ip() {
     echo 127.0.0.1
 }
 
+function _port() {
+    ${_cli} port --prefix $_prefix "$@"
```
Comment (davidvossel) [inline cluster/ephemeral-provider-common.sh]: I understand what you're doing here, but i wish we didn't have to execute a docker container to get every port. An optimization would be to run the container once and populate all the ports into env-vars.

Nothing needs to be done about this right now though i suppose.

File: cluster/ephemeral-provider-common.sh (hunk 1 above)
Comment (rmohr) [reply cluster/ephemeral-provider-common.sh]: fixed

File: cluster/ephemeral-provider-common.sh (hunk 2 above)
Comment (rmohr) [reply cluster/ephemeral-provider-common.sh]: It is really fast, so I want to keep it right now.

**Pull request merged**